### PR TITLE
Merge `package:fake_async`

### PR DESCRIPTION
**Pull request by mosuem**
- [x] Move and fix workflow files and badges in the README.md
- [x] Rev the version of the package, so that pub.dev points to the correct site
- [x] Add a line to the changelog:
```
* Move to `dart-lang/test` monorepo.
```

- [x] Add the package to the top-level readme of the monorepo:
```
| [fake_async](pkgs/fake_async/) | Fake asynchronous events such as timers and microtasks for deterministic testing. | [![pub package](https://img.shields.io/pub/v/fake_async.svg)](https://pub.dev/packages/fake_async) |
```

- [ ] **Important!** Merge the PR with 'Create a merge commit' (enabling then disabling the `Allow merge commits` admin setting)
- [ ] Update the auto-publishing settings on https://pub.dev/packages/fake_async/admin
- [ ] Add the following text to https://github.com/dart-lang/fake_async/:'

```
> [!IMPORTANT]  
> This repo has moved to https://github.com/dart-lang/test/tree/main/pkgs/fake_async
```

- [ ] Publish using the autopublish workflow
- [ ] Push tags to GitHub using `git tag --list 'fake_async*' | xargs git push origin`
- [ ] Close open PRs in dart-lang/fake_async with the following message:

`Closing as the [dart-lang/fake_async](https://github.com/dart-lang/fake_async) repository is merged into the [dart-lang/test](https://github.com/dart-lang/test) monorepo. Please re-open this PR there!`
      
- [ ] Transfer issues by running `dart run pkgs/repo_manage/bin/report.dart transfer-issues --source-repo dart-lang/fake_async --target-repo dart-lang/test --add-label package:fake_async --apply-changes`
- [ ] Archive https://github.com/dart-lang/fake_async/

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/fake_async/LICENSE [hunk 1]
```diff
@@ -0,0 +1,202 @@
+
```
Comment (jakemac53) [inline pkgs/fake_async/LICENSE:1]: Does it matter that this is not our standard license?

Comment (mosuem) [reply pkgs/fake_async/LICENSE:1]: @devoncarew ?

Comment (devoncarew) [reply pkgs/fake_async/LICENSE:1]: Let's keep the existing license as part of the move. I believe we can re-license it - into our std. license - but for here let's just do a mechanical move.

**Review by natebosch — APPROVED** [review]

**Review by jakemac53 — APPROVED** [review]

**Comment by mosuem** [comment]
To merge this (and #2293 and #2292), `Allow merge commits` needs to be enabled in this repo.

**Comment by jakemac53** [comment]
@mosuem should be enabled now

**Pull request merged**